taf/tests/test_updater/test_update/test_update_valid.py

```diff
@@ -639,3 +643,96 @@ def test_update_partial_with_invalid_commits(origin_auth_repo, client_dir):
         client_dir,
         force=True,
     )
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+        },
+    ],
+    indirect=True,
+)
+def test_update_with_removed_commits_in_auth_repo(origin_auth_repo, client_dir):
+    # Clone the repositories
+    clone_repositories(origin_auth_repo, client_dir)
+
+    # Add valid commits to the authentication repository
+    setup_manager = SetupManager(origin_auth_repo)
+    setup_manager.add_task(add_valid_target_commits)
+    setup_manager.execute_tasks()
+
+    # Remove one or more commits from the authentication repository
+    client_auth_repo = AuthenticationRepository(client_dir, origin_auth_repo.name)
+    client_target_repos = load_target_repositories(client_auth_repo)
+    remove_commits(
+        auth_repo=client_auth_repo,
+        target_repos=client_target_repos,
+        repo_path=str(client_auth_repo.path),
+        num_commits=1,
+    )
+
+    update_and_check_commit_shas(
+        OperationType.UPDATE,
+        origin_auth_repo,
+        client_dir,
+    )
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+        },
+    ],
+    indirect=True,
+)
+def test_update_with_last_validated_commit_not_in_local_repo(
+    origin_auth_repo, client_dir
+):
+    # Clone the repositories
+    clone_repositories(origin_auth_repo, client_dir)
+
+    # Push new commits to the origin repo
+    setup_manager = SetupManager(origin_auth_repo)
+    setup_manager.add_task(add_valid_target_commits)
+    setup_manager.execute_tasks()
+
+    # Manually set the last validated commit to one of the new commits
+    client_auth_repo = AuthenticationRepository(client_dir, origin_auth_repo.name)
+    last_commit_sha = client_auth_repo.last_validated_commit
+    client_target_repos = load_target_repositories(client_auth_repo)
+    origin_auth_repo.set_last_validated_commit(last_commit_sha)
```

Comment:
Something like this, but it's the client's repo whose last validated commit should be updated. The updater is used to update the client's repositories and sync them with the origin repos. Think of origin repos hosted by GitHub.